http://cygwin.com/ml/cygwin-xfree/2009-01/msg00077.html

Instead of another .cygport-specific hack, we can use the existing
BUILD_DATE constant to distinguish between releases of the same version.

--- origsrc/xorg-server-1.5.3/hw/xwin/Makefile.am	2009-01-18 22:18:25.942000000 -0600
+++ src/xorg-server-1.5.3/hw/xwin/Makefile.am	2009-01-18 22:37:10.660750000 -0600
@@ -152,7 +152,7 @@ XWin_LDADD = $(XWIN_LIBS) $(XSERVER_LIBS
 XWin_LDFLAGS = -mwindows
 
 .rc.o:
-	$(WINDRES) --use-temp-file -i $< --input-format=rc -o $@ -O coff -I $(top_builddir)/include -DPROJECT_NAME=\"$(VENDOR_NAME_SHORT)\" -DPROJECT_VERSION=\"$(PACKAGE_VERSION)\"
+	$(WINDRES) --use-temp-file -i $< --input-format=rc -o $@ -O coff -I $(top_builddir)/include -DPROJECT_NAME=\"$(VENDOR_NAME_SHORT)\" -DPROJECT_VERSION=\"$(PACKAGE_VERSION)\" -DBUILD_DATE=\"$(BUILD_DATE)\"
 
 winprefsyacc.h: winprefsyacc.c
 winprefslex.c: winprefslex.l winprefsyacc.c winprefsyacc.h
@@ -164,6 +164,7 @@ AM_YFLAGS = -d
 AM_LFLAGS = -i
 AM_CFLAGS = -DHAVE_XWIN_CONFIG_H $(DIX_CFLAGS) \
             $(XWINMODULES_CFLAGS) \
+            -DBUILD_DATE=$(BUILD_DATE) \
             -DXFree86Server
 
 dist_man1_MANS = XWin.man
--- origsrc/xorg-server-1.5.3/hw/xwin/XWin.rc	2009-01-18 22:18:23.863875000 -0600
+++ src/xorg-server-1.5.3/hw/xwin/XWin.rc	2009-01-19 02:54:14.113875000 -0600
@@ -45,7 +45,7 @@ FONT 8, "MS Shell Dlg 2"
 BEGIN
   CONTROL				IDI_XWIN, IDC_STATIC, "Static", SS_ICON, 8, 8, 32, 32
   LTEXT			PROJECT_NAME " X Server ", IDC_STATIC, 36, 8, 220, 8
-  LTEXT			"Version " PROJECT_VERSION, IDC_STATIC, 36, 18, 220, 8
+  LTEXT			"Version " PROJECT_VERSION " (" BUILD_DATE ")", IDC_STATIC, 36, 18, 220, 8
   LTEXT			"This product is licensed under the terms of the MIT/X11 License.", IDC_STATIC, 36, 38, 220, 20
   CONTROL               __VENDORDWEBSUPPORT__, ID_ABOUT_WEBSITE, "Button",
                         BS_OWNERDRAW | WS_TABSTOP, 36, 58, 68, 8
--- origsrc/xorg-server-1.5.3/hw/xwin/winerror.c	2009-01-18 22:18:24.613875000 -0600
+++ src/xorg-server-1.5.3/hw/xwin/winerror.c	2009-01-18 22:36:45.317000000 -0600
@@ -139,14 +139,14 @@ winMessageBoxF (const char *pszError, UI
 #define MESSAGEBOXF \
 	"%s\n" \
 	"Vendor: %s\n" \
-	"Release: %d.%d.%d.%d\n" \
+	"Release: %d.%d.%d.%d (%d)\n" \
 	"Contact: %s\n" \
 	"XWin was started with the following command-line:\n\n" \
 	"%s\n"
 
   pszMsgBox = Xprintf (MESSAGEBOXF,
 	   pszErrorF, VENDOR_STRING,
-		       XORG_VERSION_MAJOR, XORG_VERSION_MINOR, XORG_VERSION_PATCH, XORG_VERSION_SNAP,
+		       XORG_VERSION_MAJOR, XORG_VERSION_MINOR, XORG_VERSION_PATCH, XORG_VERSION_SNAP, BUILD_DATE,
 		       VENDOR_CONTACT,
 	   g_pszCommandLine);
   if (!pszMsgBox)
--- origsrc/xorg-server-1.5.3/hw/xwin/winprocarg.c	2009-01-18 22:18:10.051375000 -0600
+++ src/xorg-server-1.5.3/hw/xwin/winprocarg.c	2009-01-18 22:33:43.473250000 -0600
@@ -1550,7 +1550,7 @@ winLogVersionInfo (void)
 
   ErrorF ("Welcome to the XWin X Server\n");
   ErrorF ("Vendor: %s\n", VENDOR_STRING);
-  ErrorF ("Release: %d.%d.%d.%d (%d)\n", XORG_VERSION_MAJOR, XORG_VERSION_MINOR, XORG_VERSION_PATCH, XORG_VERSION_SNAP, XORG_VERSION_CURRENT);
+  ErrorF ("Release: %d.%d.%d.%d (%d)\n", XORG_VERSION_MAJOR, XORG_VERSION_MINOR, XORG_VERSION_PATCH, XORG_VERSION_SNAP, BUILD_DATE);
   ErrorF ("Contact: %s\n", VENDOR_CONTACT);
 }
 
